### DISCUSSION TITLE
Contribute to dbeaver

### DISCUSSION THREAD

**Discussion by Sargul** [dbeaver/dbeaver#571] [Q&A] [answered]
Hi,

I want to contribute to dbeaver for various database modules.

However I've not managed to get a working eclipse workspace.

I've checked out dbeaver with the m2e Eclipse Plugin from Github. However I get about 1.8k Errors in the compile process. When I use maven command line the project compiles just fine. However I can't use Eclipse to debug and start dbeaver.

I've tried:
*) checkout with m2e and egit provider
*) checkout with git command line, mvn clean install, then import maven project into eclipse

The result was always the same, 1.8k Errors. Missing dependencies to maven libraries ...

So, my question would be. What kind of settings and development process are you using.

A simple step by step guide would be highly appreciated.

As you might have guessed by now, I have little to no experience with maven build processes but I have over 17 years experience in database administration for DB2 zOS, DB2 LUW, Oracle ...

Thanks in advance

Kind Regards

Charly

**Comment by serge-rider** [comment, marked as answer]
Hi,
Glad to hear that :+1: 

DBeaver depends on a few extra plugins.
I've just published Wiki page [Develop in Eclipse](https://github.com/serge-rider/dbeaver/wiki/Develop-in-Eclipse) with details.
If you have any extra questions - please ask them in this issue.

**Comment by Sargul** [comment]
Hi Serge,

thanks for the documentation. 

Regards

Charly
